test/validationRules/required-fields.js

```diff
@@ -48,6 +57,15 @@ const requiredFieldsTestCases = {
         }
       ]
     },
+    {
+      code: "const x = gql`query { stories { id ... on Story { comments } } }`",
+      errors: [
+        {
+          message: `'comments' field must have a selection since type '[Comment]' is a Composite Type`,
+          type: "TaggedTemplateExpression"
+        }
+      ]
+    },
```

Comment:
I don't know if this one is required since in facts it test the same thing than the more simple one.

I just added it while making the test in TDD. I can remove it if you think it's not necessary.